Add transcript fields to podcast & videos both in the graphql & within monorail website

B77Mills opened the pull request:
To be done in conjunction with https://github.com/parameter1/base-platform/pull/183 && most likely a site specific site.  
 - Add transcript field to podcast and videos.
 - Create marko-web content element component for transcript field.
 - Add to Video structured data
 - Add to Podcast structured data
 
 
<img width="1015" alt="Screen Shot 2022-11-10 at 10 34 59 AM" src="https://user-images.githubusercontent.com/3845869/201153672-017ac88a-2691-45ff-b73b-99f3c1adaa67.png">
<img width="533" alt="Screen Shot 2022-11-10 at 10 35 09 AM" src="https://user-images.githubusercontent.com/3845869/201153673-270fc393-9681-4286-b1fe-747575652b04.png">

B77Mills committed:
Add trascript fields to podcat & videos.

B77Mills committed:
add transcript to webinars

B77Mills committed:
Add marko web transcript element

B77Mills committed:
Add transcript to metadata

B77Mills committed:
add transcript to google structured data

B77Mills committed:
Revert metadata additional & on add to google structured data

Revert the change to add metadata and instead only add this to google strucutred data.

B77Mills committed:
add styling for new transcript button and element

Add simple gray scale button for transcript jump link button & simple element header style for transcript element on content pages.

B77Mills committed:
Convert transcript html to plain text for structured data

B77Mills committed:
add cheerio to required pacakges

B77Mills committed:
remove transcript from metadata

solocommand commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Support for video transcripts](https://trello.com/c/udmB4Lcn/162-support-for-video-transcripts)

solocommand commented on packages/marko-web/components/page/metadata/content.marko:
```diff
@@ -33,10 +33,12 @@ fragment ContentPageMetadataFragment on Content {
   ... on ContentPodcast {
     fileSrc
     fileName
+    transcript
```
This should also be added to `ContentWebinar` type

solocommand commented on packages/marko-web/package.json:
```diff
@@ -26,6 +26,7 @@
     "@parameter1/base-cms-tenant-context": "^3.0.0",
     "@parameter1/base-cms-utils": "^3.0.0",
     "@parameter1/base-cms-web-common": "^3.3.1",
+    "cheerio": "^0.22.0",
```
Use one of the versions that are already installed in this project, `cheerio@^1.0.0-rc.10` or `cheerio@^1.0.0-rc.2`. This should have caused changes to the `yarn.lock` file, which I'm not seeing included here either.

solocommand commented on packages/marko-web/components/page/metadata/google-structured-data/content.js:
```diff
@@ -66,6 +73,7 @@ module.exports = (node) => {
       '@type': 'PodcastEpisode',
       headline: get(node, 'metadata.title'),
       ...(associatedMedia && { associatedMedia }),
+      ...(get(node, 'transcript') && { transcript: convertHtmlToText(get(node, 'transcript')) }),
```
Missing field for `Webinar`s

B77Mills replied:
@solocommand This was intentional until we have clearer direction into what type of object a webinar needs to be and what fields need to be add along with the new transcript field.

solocommand commented on packages/marko-web/components/page/metadata/google-structured-data/content.js:
```diff
@@ -49,6 +55,7 @@ module.exports = (node) => {
       uploadDate: publishedISOString,
       contentUrl: get(node, 'siteContext.canonicalUrl'),
       embedUrl: get(node, 'embedSrc'),
+      ...(get(node, 'transcript') && { transcript: convertHtmlToText(get(node, 'transcript')) }),
```
What happens if the user puts in a bad or empty value? If this throws, does the entire page error or does the JSON+LD data break quietly?

B77Mills replied:
@solocommand  I added a value check to the converthHtmlToSring function & I also added a check to make sure there is a returned value.  In testing or trying to make it break I was only able to make cheerio break by not sending a value at all.  It converted broken or crappy html just fine.  I can add some additional check but not sure if you want me to wrap the cheerio.load() in some sort of try catch?  Thoughts?

B77Mills committed:
Update version of cheerio being included

B77Mills committed:
add tanscirpt to webinar fragment

B77Mills committed:
Update convertHtmlToString functino with value check

Add check to converter to ensure there is a value to convert.  Also check that there is a value being returned and only display when there is.

B77Mills committed:
fix problem with return value
